### DISCUSSION TITLE
How can I make or publish an Embedded program by Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11527] [Q&A] [answered]
Recently I found a video on YouTube about Avalonia UI Industrial Embedded System: https://www.youtube.com/watch?v=LOxMWhYcKqA&ab_channel=DanWalmsley

I guess it may be a Linux program. However, almost all the Linux programs made by Avalonia I have ever seen need a graphical desktop and must run after the graphical desktop has booted.

I want to boot my program when I press the power button as the video showed and run it without a graphical desktop.

How can I achieve this?

**Comment by rabbitism** [comment, marked as answer]
https://docs.avaloniaui.net/guides/deep-dives/running-on-raspbian-lite-via-drm

You can check this tutorial.

**Reply by kyuranger** [reply]
> https://docs.avaloniaui.net/guides/deep-dives/running-on-raspbian-lite-via-drm
> 
> You can check this tutorial.

It seems this tutorial still need a graphical desktop to run.

**Reply by rabbitism** [reply]
> > https://docs.avaloniaui.net/guides/deep-dives/running-on-raspbian-lite-via-drm
> > 
> > You can check this tutorial.
> 
> It seems this tutorial still need a graphical desktop to run.

Why? Pi lite os is not shipped with desktop environment.

**Reply by kyuranger** [reply]
> > > https://docs.avaloniaui.net/guides/deep-dives/running-on-raspbian-lite-via-drm
> > > You can check this tutorial.
> > 
> > 
> > It seems this tutorial still need a graphical desktop to run.
> 
> Why? Pi lite os is not shipped with desktop environment.

I am beginning of Linux but it seems running a GUI program must need a desktop environment (just like GNOME).

**Reply by maxkatz6** [reply]
It renders directly to the Framebuffer, there is no desktop.

**Reply by kyuranger** [reply]
> It renders directly to the Framebuffer, there is no desktop.

Let me try this. Thank you.
